util/cosmos/iscn/sign.js

```diff
@@ -125,6 +124,79 @@ function formatISCNPayload(payload, version = 1) {
   };
 }
 
+export async function estimateISCNTxFee(tx, {
+  version = 1,
+} = {}) {
+  const record = formatISCNPayload(tx);
+  const feePerByte = await queryFeePerByte();
+  const feePerByteAmount = feePerByte ? parseInt(feePerByte.amount, 10) : 1;
+  const {
+    recordNotes,
+    contentFingerprints,
+    stakeholders,
+    contentMetadata,
+  } = record;
+  const now = new Date();
+  const obj = {
+    '@context': {
+      '@vocab': 'http://iscn.io/',
+      recordParentIPLD: {
+        '@container': '@index',
+      },
+      stakeholders: {
+        '@context': {
+          '@vocab': 'http://schema.org/',
+          entity: 'http://iscn.io/entity',
+          rewardProportion: 'http://iscn.io/rewardProportion',
+          contributionType: 'http://iscn.io/contributionType',
+          footprint: 'http://iscn.io/footprint',
+        },
+      },
+      contentMetadata: {
+        '@context': null,
+      },
+    },
+    '@type': 'Record',
+    '@id': `iscn://${ISCN_REGISTRY_NAME}/btC7CJvMm4WLj9Tau9LAPTfGK7sfymTJW7ORcFdruCU/1`,
+    recordTimestamp: now.toISOString(),
+    recordVersion: version,
+    recordNotes,
+    contentFingerprints,
+    recordParentIPLD: {},
+  };
+  if (version > 1) {
+    obj.recordParentIPLD = {
+      '/': 'bahuaierav3bfvm4ytx7gvn4yqeu4piiocuvtvdpyyb5f6moxniwemae4tjyq',
+    };
+  }
+
+  const byteSize = Buffer.from(jsonStringify(obj), 'utf-8').length
+    + stakeholders.reduce((acc, s) => acc + s.length, 0)
+    + contentMetadata.length;
+  const feeAmount = byteSize * feePerByteAmount;
+
+  return feeAmount;
+}
+
+export async function estimateISCNTxGas(tx, version) {
+  const DEFAULT_GAS = 1000000; // TODO: estimate according to size
```

Comment:
Please refactor this to `ISCN_GAS` to `@/constant`, also would `200000` be enough?

Reply:
I think so. So far my test tx gas_used are all < 150,000 nanolike